### ISSUE TITLE
FetchError: request to URL/login failed, reason: self signed certificate in certificate chain

### ISSUE THREAD

**Issue by styrus** [jest-community/vscode-jest#592]
I am using Frisby framework for testing my APIs along with JEST as my test tool.
I am facing issues **self signed certificate in certificate chain** whenever I try to run my TC usuing JEST.

**_Note: My same test below passes when run with jasmine and serial-jasmine_**
I have defined node property to disable ssl

`process.env.NODE_TLS_REJECT_UNAUTHORIZED = "0"; 
`
also using 
```
strictSSL : false,
rejectUnauthorized: false-
```
Here is my code snippet.

```
process.env.NODE_TLS_REJECT_UNAUTHORIZED = "0"; 
var frisby = require('frisby.js');
var global= require("../../../global");
jasmine.DEFAULT_TIMEOUT_INTERVAL = 20000;



it ('setup method',  function () {
    return frisby
    .setup({
        request:{
            headers: {
                'Authorization': 'Basic ' + Buffer.from("usrname:password").toString('base64'),
                'headers':'value',
                'Content-Type': 'application/json'
            },
            strictSSL : false,
            rejectUnauthorized: false
        }
      })
      
    .get(global.urls.baseUrl+"/token")
    .inspectResponse()
    .expect('status', 200);

});

```
my Jest settings in package.json
 ```
"scripts": {
    "test": "jest"
  },
```
Can any one please point how can I resolve this _"self signed certificate in certificate chain"_
It would be great. Thanks.

**Comment by styrus** [comment]
@connectdotz 
Can you please suggest anything on this. Would be great if you pinpoint the resolution.

**Comment by connectdotz** [comment]
are you able to run jest in a terminal?

**Comment by G4gan** [comment]
@connectdotz 
Yes I am able to run other test for some methods (created calculator functions) and calling them in my test file.
This works when run through JEST.

Just when running api calls with HTTPS. it errors this
**FetchError: request to URL/login failed, reason: self signed certificate in certificate chain**

**Comment by connectdotz** [comment]
looks like this is an open issue on jest side: facebook/jest#8449. `vscode-jest` just parse jest test run results, so first jest has to be able to run your HTTP tests at least on the terminal, i.e. outside of vscode.

**Comment by styrus** [comment]
Thanks @connectdotz.
Yes I am able to run other testcases using JEST just getting this error when running test with https url.
Do you have any info on when this issue will be resolved. The one you mention above.

**Comment by cipher-code** [comment]
Your API server is using a self signed certificate, which is being rejected by node. While the issue may seem on the surface to be a jest issue, its actually node that is throwing the error.

There are a number of ways to get node to trust the certificate such as monkey patching your operating systems TLS but I wouldn't recommend them due to these methods being overly complex to setup and security reasons.

If you have your API running locally, one easy way around this is to use [expose.sh](https://expose.sh?utm_source=githubIssues).

You can run `expose 80` then it'll generate a public HTTPs URL which will forward to localhost that you can use for testing your API, like `https://a8kc.expose.sh`.

I've written a step by step guide [here](https://expose.sh/blog/2020/07/11/local-https-testing?utm_source=githubIssues).

**Comment by neonbeast** [comment]
Sign in failed: FetchError: request to https://vscode-auth.github.com/token?code=ce1efdca39ac3fd967ea&state=c5abfe0d-49c0-41fd-9485-312c7d34e8e2 failed, reason: connect EAFNOSUPPORT 54.235.211.105:443 - Local (0.0.0.0:49407)



pls help
